vscode/src/completions/text-processing.ts

```diff
@@ -167,20 +157,9 @@ export function trimUntilSuffix(insertion: string, prefix: string, suffix: strin
             line = prefix.slice(lastNewlineOfPrefix + 1) + line
         }
 
-        /**
-         * We cut the completion on the partial match of the suffix
-         *
-         * For example, if the suffix is:
-         * `  key: ${{ runner.os }}-pnpm-store-${{ hashFiles('pnpm-lock.yaml') }}`
-         *
-         * And the completions is:
-         * `pnpm-store\n  key: ${{ runner.os }}-pnpm-${{ steps.pnpm-cache.outputs.STORE_PATH }}`
-         *
-         * We cut the completion on the `  key:` part to avoid duplicating the suffix.
-         * The resulting completion will be: `pnpm-store`.
-         */
-        const linePart = getFirstNCharsPreservingLeadingSpaces(line, NUMBER_OF_CHARS_TO_MATCH_AND_CUT_SUFFIX)
-        if (linePart.length && firstNonEmptySuffixLinePart.startsWith(linePart)) {
+        const isSameIndentation = indentation(line) === indentation(firstNonEmptySuffixLine)
```

Comment:
Only last nit: Should this be 

```suggestion
        const isSameIndentation = indentation(line) <= indentation(firstNonEmptySuffixLine)
```

just to make sure we have a bit more resilience against screwed up indentation haha